extensions/wikia/EditAccount/SpecialEditAccount_body.php

```diff
@@ -518,4 +529,55 @@ public static function generateRandomScrambledPassword() {
 		$REQUIRED_CHARS = "A1a";
 		return (wfGenerateToken() . $REQUIRED_CHARS);
 	}
+
+	/** Hook for storing historical log of email changes **/
+	public static function logEmailChanges($user, $new_email, $old_email) {
+		global $wgExternalSharedDB, $wgUser, $wgRequest;
+		if ( $wgExternalSharedDB ) {
+			$dbw = wfGetDB( DB_MASTER, array(), $wgExternalSharedDB );
+			$dbw->insert(
+				'user_email_log',
+				['user_id' => $user->getId(),
+				 'old_email' => $old_email,
+				 'new_email' => $new_email,
+				 'changed_by_id' => $wgUser->getId(),
+				 'changed_by_ip' => $wgRequest->getIP()  // stored as string
+				]);
+		}
+		return true;
+	}
+
+	public function displayLogData() {
+		global $wgExternalSharedDB, $wgOut, $wgRequest;
+
+		$user_id = $wgRequest->getVal('user_id', null);
+		$rows = [];
+
+		if ( $wgExternalSharedDB && $user_id ) {
+			$dbr = wfGetDB ( DB_SLAVE, array(), $wgExternalSharedDB );
+			$res = $dbr->select (
+				'user_email_log',   // from
+				["*"], 				// cols
+				"user_id = " . $dbr->strencode($user_id), // where
```

Comment:
`[ 'user_id' => $user_id ]` should be preferred for where clauses over manual escaping. `strencode` doesn't escape spaces and so forth, so this can still cause issues.